drivers/shared/executor/procstats/list_windows.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+//go:build windows
+
+package procstats
+
+import (
+	"github.com/hashicorp/go-set/v2"
+	"github.com/mitchellh/go-ps"
+)
+
+func gather(procs map[int]ps.Process, family set.Collection[int], root int, candidate ps.Process) bool {
+	if candidate == nil {
+		return false
+	}
+	pid := candidate.Pid()
+	if pid == 0 || pid == 1 {
+		return false
+	}
+	if pid == root {
+		return true
+	}
+	parent := procs[candidate.PPid()]
+	result := gather(procs, family, root, parent)
+	if result {
+		family.Insert(pid)
+	}
+	return result
+}
+
+func mapping(all []ps.Process) map[int]ps.Process {
+	result := make(map[int]ps.Process)
+	for _, process := range all {
+		result[process.Pid()] = process
+	}
+	return result
+}
+
+func list(executorPID int, processes func() ([]ps.Process, error)) set.Collection[ProcessID] {
+	family := set.From([]int{executorPID})
+
+	all, err := processes()
+	if err != nil {
+		return set.New[ProcessID](0)
```

Comment:
Could we at least return the executor PID here? Or is the thinking that if this call fails we can't trust any of the results?

Reply:
Oh good idea, let me add that